test/core/event_engine/posix/traced_buffer_list_test.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/event_engine/posix_engine/traced_buffer_list.h"
+
+#include <atomic>
+
+#include <gtest/gtest.h>
+
+#include <grpc/grpc.h>
+
+#include "src/core/lib/iomgr/port.h"
+#include "test/core/util/test_config.h"
+
+#ifdef GRPC_LINUX_ERRQUEUE
+
+#define NUM_ELEM 5
+
+namespace grpc_event_engine {
+namespace posix_engine {
+namespace {
+
+void TestShutdownFlushesListVerifier(void* arg, Timestamps* /*ts*/,
+                                     absl::Status status) {
+  ASSERT_TRUE(status.ok());
+  ASSERT_NE(arg, nullptr);
+  int* done = reinterpret_cast<int*>(arg);
+  *done = 1;
+}
+
+void TestVerifierCalledOnAckVerifier(void* arg, Timestamps* ts,
+                                     absl::Status status) {
+  ASSERT_TRUE(status.ok());
+  ASSERT_NE(arg, nullptr);
+  ASSERT_EQ(ts->acked_time.time.clock_type, GPR_CLOCK_REALTIME);
+  ASSERT_EQ(ts->acked_time.time.tv_sec, 123);
+  ASSERT_EQ(ts->acked_time.time.tv_nsec, 456);
+  ASSERT_GT(ts->info.length, 0);
+  int* done = reinterpret_cast<int*>(arg);
+  *done = 1;
+}
+
+}  // namespace
+
+// Tests that all TracedBuffer elements in the list are flushed out on shutdown.
+// Also tests that arg is passed correctly.
+TEST(BufferListTest, Testshutdownflusheslist) {
```

Comment:
Test names are conventionally written in CamelCase.

Reply:
Fixed.